apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt

```diff
@@ -264,4 +264,24 @@ class ConfigurationTests {
       assertThat(transformedQuery.readText(), containsString("__typename"))
     }
   }
-}
\ No newline at end of file
+
+  @Test
+  fun `compilation unit directory properties carry task dependencies`() {
+    withSimpleProject("""
+      apollo {
+        generateTransformedQueries = true
+        
+        compilationUnits.all { compilationUnit ->
+          def customTaskProvider = tasks.register("customTask" + compilationUnit.name) {
```

Comment:
Side note: `name` is a bit clunky to use since it's comprised of '`variant name` + `service name`', but neither are capitalized

cc @martinbonnin

Reply:
`name` is really only used to make the `NamedDomainObjectContainer` happy but it's not really used. If there's another way to do `compilationUnits.all {}` without having to give CompilationUnit a `name`, I'd prefer this. I only went for `NamedDomainObjectContainer` bevause it's ubiquitous in the gradle code I've seen so far but it might be overkill.

Reply:
Gotcha, forgot the container requires a name (an override property would this more obvious, thanks Groovy)